porcupine/plugins/markdown.py

```diff
@@ -0,0 +1,56 @@
+"""If configuration says so, insert spaces when the tab key is pressed."""
+
+from __future__ import annotations
+
+import logging
+import re
+import tkinter
+from functools import partial
+
+from porcupine import get_tab_manager, tabs, textutils, utils
+
+log = logging.getLogger(__name__)
+
+
+setup_before = ["tabs2spaces"]
+
+
+def _is_list_item(line: str) -> bool:
+    """Detect if the line that is passed is a markdown list item
+
+    According to:
+    - https://spec.commonmark.org/0.30/#lists
+    - https://pandoc.org/MANUAL.html#lists
+    """
+    assert len(line.splitlines()) == 1
```

Comment:
I sometimes get an error from this line:

```
porcupine._state ERROR: Error in tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
  File "/home/akuli/porcu/porcupine/utils.py", line 548, in callback
    return on_tab(event, shifted)
  File "/home/akuli/porcu/porcupine/plugins/markdown.py", line 38, in on_tab_key
    list_item_status = _is_list_item(line)
  File "/home/akuli/porcu/porcupine/plugins/markdown.py", line 25, in _is_list_item
    assert len(line.splitlines()) == 1
AssertionError
```

reason:

```
>>> "".splitlines()
[]
```

Reply:
Interesting.. Do you have a way to reproduce that?

Reply:
I got it when I tried to do something else, but I guess pressing tab on a blank line does it.